docs/deployment_design/design.md

```diff
@@ -0,0 +1,63 @@
+# Llama index deployment design
+
+## Component diagram
+
+![Alt text](components.png)
+
+## Deployment diagram
+
+![Alt text](deployment.png)
+
+## LlamaIndex API startup sequence diagram
+
+```mermaid
+sequenceDiagram
+    participant creator as App Creator
+
+    create participant sc as :StorageContext
+    creator->>sc: create with index_folder
+    create participant index as :BaseIndex
+    creator->>index: create with storage_context
+    create participant retriever as :VectorIndexRetriever
+    creator->>retriever: create with index
+    create participant rqe as :RetrieverQueryEngine
+    creator->>rqe: create with retriever
+    create participant conf as :Config
+    creator->>conf: create with query_engine
+    create participant app as :Flask
+    creator->>app: create with config
+
+```
```

Comment:
The design looks great. It's really fast to pick how you plan to integrate the project. It's just the start-up sequence diagram that is a bit unclear to me. Maybe because I haven't seen a lot of those. Am I understanding right that when the app starts, it goes like this:
- `App Creator` creates a `StorageContext`
- `StorageContext` creates a `BaseIndex`
- And so on until we arrive at the `Flask app` ready to process the queries

Reply:
Yes pretty much it, except `App Creator` creates all those objects, one after the other, passing the previous to the constructor of the next. I decided to make a sequence diagram of the startup process because it will take longer than a regular flask app startup. Could be helpful to know the steps that cause the delay. I will explain that in the section.

Reply:
So that's why the arrows always start at `App Creator`, I understand better now.